resource_calendar_flexible_exclude_weekend/models/resource_calendar.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+from collections import defaultdict
+from datetime import timedelta
+
+from dateutil.rrule import DAILY, rrule
+from pytz import timezone
+
+from odoo import fields, models
+
+from odoo.addons.hr_work_entry_contract.models.hr_work_intervals import WorkIntervals
+
+_logger = logging.getLogger(__name__)
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    exclude_weekends = fields.Boolean()
+
+    # Override to return weekends as special days if exclude_weekends is set
+    def _attendance_intervals_batch(
+        self, start_dt, end_dt, resources=None, domain=None, tz=None, lunch=False
+    ):
+        """
+        Override to adjust start_dt if it falls on a weekend.
+        - Convert start_dt to the relevant timezone
+        - If Saturday or Sunday → move to next Monday
+        - Then call super() with the adjusted start_dt
+        """
+        # Ensure timezone awareness
+        assert start_dt.tzinfo and end_dt.tzinfo, "Datetimes must be timezone-aware"
+        self.ensure_one()
+
+        if not resources:
+            resources = self.env["resource.resource"]
+            resources_list = [resources]
```

Comment:
```suggestion
            resources_list = resources
```

Why transforming this empty recordset into a list ?

Reply:
After looking into it further, it appears to originate from the original method defined in the resource addon. It might be better to simplify the logic rather than copying the complex code from there.